### PR TITLE
more within try block. escape regex special chars

### PR DESCRIPTION
**Pull request by amunger**
Fixes https://github.com/microsoft/vscode-jupyter/issues/13999

### CODE REVIEW

File: src/extension.node.ts
Comment (Tyriar) [inline src/extension.node.ts:253]: This seems pretty fragile, can we use indexOf/contains instead?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/extension.node.ts
Comment (amunger) [reply src/extension.node.ts:253]: to escape each occurrence of a regex special character? That sounds more fragile

Comment (Tyriar) [reply src/extension.node.ts:253]: I mean do plain string comparisons without touching regex at all. If you replace one of these special characters isn't this logic going to fall over? Posix paths can have many more characters than Windows paths for example which could be part of `$HOME`.

Comment (amunger) [reply src/extension.node.ts:253]: I think it would still work after escaping the character, I'm not removing them.
But yeah, I can try to manually find and replace all occurrences, still doubt it will be less fragile than just using regex.

Comment (Tyriar) [reply src/extension.node.ts:253]: Up to you, I approved if you don't think it's worth it

**Review by Tyriar — APPROVED** [review]

File: src/extension.node.ts
Comment (amunger) [reply src/extension.node.ts:253]: I'll let @DonJayamanne take a look, since he actually owns the logging area

**Pull request merged**
